tests/playwright/specs/a11y/components.spec.js

```diff
@@ -0,0 +1,177 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+'use strict';
+
+const openPage = require('../../scripts/open-page');
+const { createReport, scanPageElement } = require('./includes/helpers');
+const {
+    navigationLocator,
+    footerLocator,
+    subNavigationLocator
+} = require('./includes/locators');
+const { test, expect } = require('@playwright/test');
+
+const testURL = '/en-US/firefox/new/';
+
+test.describe(
+    'Navigation (desktop)',
+    {
+        tag: '@a11y'
+    },
+    () => {
+        test.beforeEach(async ({ page, browserName }) => {
+            await openPage(testURL, page, browserName);
+        });
+
+        test('should not have any detectable a11y issues', async ({ page }) => {
+            const firefoxLink = page.getByTestId('navigation-link-firefox');
+            const firefoxMenu = page.getByTestId('navigation-menu-firefox');
+
+            // Hover over Firefox link to open menu
+            await expect(firefoxMenu).not.toBeVisible();
+            await firefoxLink.hover();
+            await expect(firefoxMenu).toBeVisible();
+
+            const results = await scanPageElement(page, navigationLocator);
+            createReport('component', 'navigation-desktop', results);
+            expect(results.violations.length).toEqual(0);
+        });
+    }
+);
+
+test.describe(
+    'Navigation (mobile)',
+    {
+        tag: '@a11y'
+    },
+    () => {
+        test.use({ viewport: { width: 360, height: 780 } });
+
+        test.beforeEach(async ({ page, browserName }) => {
+            await openPage(testURL, page, browserName);
+        });
+
+        test('should not have any detectable a11y issues', async ({ page }) => {
+            const navigationMenuButton = page.getByTestId(
+                'navigation-menu-button'
+            );
+            const navigationMenuItems = page.getByTestId(
+                'navigation-menu-items'
+            );
+            const firefoxLink = page.getByTestId('navigation-link-firefox');
+            const firefoxMenu = page.getByTestId('navigation-menu-firefox');
+
+            // Open navigation menu
+            await expect(navigationMenuItems).not.toBeVisible();
+            await navigationMenuButton.click();
+            await expect(navigationMenuItems).toBeVisible();
+
+            // Open Firefox menu
+            await expect(firefoxMenu).not.toBeVisible();
+            await firefoxLink.click();
```

Comment:
**Enhancement/Thought:** Since these are a11y tests it would be good to test that keyboard navigation works but it's not a blocker for this PR.

Reply:
I'd probably consider those to be functional tests as opposed to something specific to the a11y test job, but it would be a good bit of extra coverage agreed 👍